src/mon/OSDMonitor.cc

```diff
@@ -9072,10 +9072,6 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	 << " < firefly, which is required for primary-temp";
       err = -EPERM;
       goto reply;
-    } else if (!g_conf->mon_osd_allow_primary_temp) {
```

Comment:
> These are covered by require_osd_release in the OSDMap.

i understand that OSDs are required to be all luminous+ in post luminous. but how can we educate old clients to understand this?

Reply:
These options are superceded by the new osdmap require-min-compat-client which controls which newish incompat things are allowed to be used.